### PR TITLE
#20: Fixed LuaCheck findings

### LINKED ISSUE #20: Split SQL Renderer
## Situation

At the moment the complete SQL rendering is in one large module. This is not sustainable. We need to split the renderer into smaller, more manageable pieces.

## Acceptance criteria

1. Renderer split
2. All test still green

### PR DESCRIPTION
**Pull request by redcatbear**
Closes #20.

I created a couple of issues for missing renderer coverage in order to not make this PR any bigger.

### CODE REVIEW

File: src/exasolvs/queryrenderer/ScalarFunctionAppender.lua [hunk 1]
```diff
@@ -0,0 +1,313 @@
...
+local ScalarFunctionAppender = {}
+ScalarFunctionAppender.__index = ScalarFunctionAppender
+setmetatable(ScalarFunctionAppender, {__index = AbstractQueryAppender})
+
+--- Create a new instance of a `ScalarFunctionRenderer`.
+-- @param out_query query to which the function will be appended
+-- @return renderer for scalar functions
+function ScalarFunctionAppender:new(out_query)
+    assert(out_query ~= nil,
+            "Renderer for scalar function requires a query object that it can append to.")
```
Comment (kaklakariada) [inline src/exasolvs/queryrenderer/ScalarFunctionAppender.lua:15]: Could you avoid duplication by moving this check to `AbstractQueryAppender._init(self, out_query)`?

File: src/exasolvs/queryrenderer/ScalarFunctionAppender.lua [hunk 2]
```diff
@@ -0,0 +1,313 @@
...
+end
+
+function ScalarFunctionAppender:_append_simple_function(f)
+    self:_append(string.upper(f.name))
+    self:_append_function_argument_list(f.arguments)
+end
+
+-- Numeric functions
+ScalarFunctionAppender._abs = ScalarFunctionAppender._append_simple_function
+ScalarFunctionAppender._acos = ScalarFunctionAppender._append_simple_function
```
Comment (kaklakariada) [inline src/exasolvs/queryrenderer/ScalarFunctionAppender.lua:75]: 👍

**Review by kaklakariada — CHANGES_REQUESTED** [review]
Please create a ticket to render LuaDoc and use `@tparam` and `@treturn`tags

File: spec/queryrenderer/ExpressionAppender_spec.lua [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+        it("string", function()
+            assert_expression_yields(literal.string("hello world"), "'hello world'")
+        end)
+
+        it("date", function()
+            assert_expression_yields(literal.date("2022-12-31"), "DATE '2022-12-31'")
+        end)
+
+        it("timestamp", function()
+            assert_expression_yields(literal.timestamp("12:30:24.007"), "TIMESTAMP '12:30:24.007'")
```
[the remark this answers is not in the document]
Comment (redcatbear) [reply spec/queryrenderer/ExpressionAppender_spec.lua:47]: Fixed.

File: src/exasolvs/Query.lua [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+--- Append a single token.
+-- While the same can be achieved with calling <code>appendAll</code> with a single parameter, this method is faster.
+-- @param token token to append
+function Query:append(token)
+    self.tokens[#self.tokens + 1] = token
+end
+
+--- Append all tokens.
+-- @param ... tokens to append
+function Query:appendAll(...)
```
[the remark this answers is not in the document]
Comment (redcatbear) [reply src/exasolvs/Query.lua:33]: Fixed.

File: src/exasolvs/queryrenderer/AbstractQueryAppender.lua [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+    end
+end
+
+function AbstractQueryAppender:_append_decimal(data_type)
+    self:_append("(")
+    self:_append(data_type.precision)
+    self:_append(",")
+    self:_append(data_type.scale)
+    self:_append(")")
+end
```
[the remark this answers is not in the document]
Comment (redcatbear) [reply src/exasolvs/queryrenderer/AbstractQueryAppender.lua:37]: Renamed.

File: src/exasolvs/queryrenderer/ExpressionAppender.lua [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+
+function ExpressionAppender:_append_exists(sub_select)
+    self:_append("EXISTS(")
+    require("exasolvs.queryrenderer.SelectAppender"):new(self.out_query):append_select(sub_select.query)
+    self:_append(")")
+end
+
+function ExpressionAppender:_append_unary_predicate(predicate)
+    self:_append("(")
+    self:_append(OPERATORS[predicate.type])
```
[the remark this answers is not in the document]
Comment (redcatbear) [reply src/exasolvs/queryrenderer/ExpressionAppender.lua:46]: Done.

File: src/exasolvs/queryrenderer/ScalarFunctionAppender.lua (hunk 1 above)
Comment (redcatbear) [reply src/exasolvs/queryrenderer/ScalarFunctionAppender.lua:15]: I could, but I want the program to terminate as early as possible. This is better pinpointing in my eyes.

File: src/exasolvs/queryrenderer/SelectAppender.lua [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+    if join_type_keyword then
+        self:_append_table(join.left)
+        self:_append(' ')
+        self:_append(join_type_keyword)
+        self:_append(' JOIN ')
+        self:_append_table(join.right)
+        self:_append(' ON ')
+        self:_append_expression(join.condition)
+    else
+        error('E-VS-QR-6: Unable to render unknown join type "' .. join.join_type .. '".')
```
[the remark this answers is not in the document]
Comment (redcatbear) [reply src/exasolvs/queryrenderer/SelectAppender.lua:68]: #24

**Review by kaklakariada — DISMISSED** [review]

**Review by kaklakariada — APPROVED** [review]

**Pull request merged**